### PR TITLE
refactor: abstract apm client to a separate package

### PR DESCRIPTION
**Pull request by kruskall**
Start working on decoupling apm and extension package.
Shutdown the apm receiver gracefully and move away from
deprecated ioutil calls.

Still untested but ready for review

### CODE REVIEW

File: apm-lambda-extension/apm/apmserver.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+// StartBackgroundApmDataForwarding Receive agent data as it comes in and post it to the APM server.
```
Comment (axw) [inline apm-lambda-extension/apm/apmserver.go:34]:
```suggestion
// ForwardApmData receives agent data as it comes in and posts it to the APM server.
```

**Review by axw — APPROVED** [review]
Thanks, loving this refactoring.

Just one minor thing: "apm" feels too generic of a package name. Perhaps call it "apmproxy", or something along those lines?

File: apm-lambda-extension/apm/apmserver.go (hunk 1 above)
Comment (kruskall) [reply apm-lambda-extension/apm/apmserver.go]: Good catch! 

I also renamed the package :+1:

**Pull request merged**